File: 4. orientation-part/week-10/day-1/data-structures/map-introduction2.js
const myMap = new Map([
	["978-1-60309-452-8", "A Letter to Jo"],
	["978-1-60309-459-7", "Lupus"],
	["978-1-60309-444-3", "Red Panda and Moon Bear"],
	["978-1-60309-461-0", "The Lab"],
]);

myMap.forEach(function (value, key) {
	console.log(key + " (" + value + ")");
});

myMap.delete("978-1-60309-444-3");
myMap.delete(getKeyByValue(myMap, "The Lab"));

myMap.forEach(function (value, key) {
	console.log(key + " (" + value + ")");
});

myMap.set("978-1-60309-450-4", "They Called Us Enemy");
myMap.set("978-1-60309-453-5", "Why Did We Trust Him?");

console.log(myMap.has("478-0-61159-424-8"));
console.log(myMap.get("978-1-60309-453-5"));

function getKeyByValue(map, value) {
	for (let [k, v] of map.entries()) {
		if (v === value) {
			return k;
		}
	}
	return null;
}

File: greenfox/1. foundation-part/data-structures/MapIntroduction2.java
import java.util.HashMap;
import java.util.Map.Entry;
public class MapIntroduction2 {
	public static void main(String[] args) {
		HashMap<String, String> myHashmap = new HashMap<>();
		myHashmap.put("978-1-60309-452-8", "A Letter to Jo");
		myHashmap.put("978-1-60309-459-7", "Lupus");
		myHashmap.put("978-1-60309-444-3", "Red Panda and Moon Bear");
		myHashmap.put("978-1-60309-461-0", "The Lab");

		for (String key: myHashmap.keySet()) {
			System.out.println(myHashmap.get(key) + " (ISBN: " + key + ")");
		}

		myHashmap.remove("978-1-60309-444-3");
		String value = "978-1-60309-461-0";

		for (Entry<String, String> entry : myHashmap.entrySet()) {
			if (entry.getValue() == value) {
				myHashmap.remove(entry.getKey());
				break;
			}
		}

		myHashmap.put("978-1-60309-450-4", "They Called Us Enemy");
		myHashmap.put("978-1-60309-453-5", "Why Did We Trust Him?");

		System.out.println(myHashmap.containsKey("478-0-61159-424-8"));
		System.out.println(myHashmap.get("978-1-60309-453-5"));
	}
}
